src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,82 @@
+'use client';
+
+import type { ComponentProps, FC } from 'react';
+import { useEffect } from 'react';
+import { twMerge } from 'tailwind-merge';
+import { mergeDeep } from '../../helpers/merge-deep';
+import { getTheme } from '../../theme-store';
+import type { DeepPartial } from '../../types';
+import { DrawerHeader, type FlowbiteDrawerHeaderTheme } from './DrawerHeader';
+import { DrawerContext } from './DrawerContext';
+import type { FlowbiteBoolean } from '../Flowbite';
+import { DrawerItems, type FlowbiteDrawerItemsTheme } from './DrawerItems';
+
+export interface FlowbiteDrawerTheme {
+  root: FlowbiteDrawerRootTheme;
+  header: FlowbiteDrawerHeaderTheme;
+  items: FlowbiteDrawerItemsTheme;
+}
+
+export interface FlowbiteDrawerRootTheme {
+  base: string;
+  position: {
+    top: string;
+    right: string;
+    bottom: string;
+    left: string;
+  };
+  collapsed: FlowbiteBoolean;
+}
+
+export interface DrawerProps extends ComponentProps<'div'> {
+  open?: boolean;
+  position?: 'top' | 'right' | 'bottom' | 'left';
+  onClose: () => void;
+  theme?: DeepPartial<FlowbiteDrawerTheme>;
+}
+
+const DrawerComponent: FC<DrawerProps> = ({
+  children,
+  className,
+  open: isOpen = false,
+  theme: customTheme = {},
+  position = 'left',
+  onClose,
+  ...props
+}) => {
+  const theme = mergeDeep(getTheme().drawer, customTheme);
+
+  useEffect(() => {
+    const handleEscapeKey = (event: KeyboardEvent) => {
```

Comment:
You would probably need to check if there is no input element within the drawer.
Because if the user has any sort of textarea/input/combobox inside the drawer, this code would close while they are typing

Reply:
but in this we're checking the specific key, isn't it?

Reply:
more exact: check if there is any input element focused, of any kind before executing any action on "ESC" key press.

Reply:
any idea how to check it @SutuSebastian ?